src/main/kotlin/com/transportes/services/DocumentService.kt

```diff
@@ -1,19 +1,42 @@
 package com.transportes.services
 
-import com.transportes.dto.DocumentDTO
+import com.transportes.dto.document.DocumentDTO
 import com.transportes.repositories.DocumentRepository
 import com.transportes.utils.Serializer
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.stereotype.Service
+import com.transportes.domain.documents.Document
+import com.transportes.domain.users.SoloCarrier
+import com.transportes.repositories.EmployeeRepository
+import com.transportes.repositories.UserRepository
 
 @Service
 class DocumentService {
     @Autowired lateinit var documentRepository: DocumentRepository
+    @Autowired lateinit var userRepository: UserRepository
+    @Autowired lateinit var employeeRepository: EmployeeRepository
 
     fun getUserDocuments(idUser: String): List<DocumentDTO> {
         val documentsList = documentRepository.findByUserId(idUser)
         return documentsList.map { document ->
             Serializer.buildDocumentDTO(document)
         }
     }
+
+    fun createDocument(dto: DocumentDTO): Document {
+        val document = Document(
+            soloCarrier = null,
+            employee = null,
+            name = dto.name,
+            linkImage = dto.fileLink
+        )
+        if (employeeRepository.existsById(dto.idUser)) {
+            document.employee = employeeRepository.findById(dto.idUser).get()
+        } else if (userRepository.existsById(dto.idUser)) {
+            document.soloCarrier = userRepository.findById(dto.idUser).get() as SoloCarrier?
+        } else {
+            throw RuntimeException("Usuario no encontrado")
```

Comment:
```suggestion
            throw NotFoundException("Usuario no encontrado")
```
Si dejas RuntimeException, el servicio lanza error 500. En este caso tenemos q lanzar 400, ya que seria un error del usuario, no de servidor.